scripts/typeGeneration/convert.js

```diff
@@ -0,0 +1,135 @@
+import { basename, dirname, join, resolve } from 'https://deno.land/std/path/posix.ts';
+import { parse, stringify } from 'https://deno.land/std/yaml/mod.ts';
+import { camelCase } from 'https://deno.land/x/case/mod.ts';
+import { generate } from 'npm:openapi-typescript-codegen';
+import converter from 'npm:swagger2openapi';
+
+const apiTypes = ['management'];
+
+const getFiles = async folder => {
+  const files = [];
+  for await (const singleDirEntry of Deno.readDir(folder)) {
+    if (singleDirEntry.isDirectory) {
+      files.push(...(await getFiles(join(folder, singleDirEntry.name))));
+    } else if (singleDirEntry.isFile) {
+      const res = resolve(folder, singleDirEntry.name);
+      const filename = res.substring(res.lastIndexOf('/') + 1, res.indexOf('.yml'));
+      const apiType = filename.substring(0, filename.indexOf('_'));
+      let versionSuffix = filename.substring(filename.lastIndexOf('_'));
+      versionSuffix = versionSuffix != '_api' ? versionSuffix : '';
+      const service = basename(dirname(res));
+      if (apiTypes.includes(apiType)) {
+        files.push({ filename, apiType, path: res, service, versionSuffix });
+      }
+    }
+  }
+  return files;
+};
+
+const getFileContents = async file => {
+  const fileContent = await Deno.readTextFile(file.path);
+  const fileData = parse(fileContent);
+  if (!fileData.swagger) {
+    return { ...file, fileData };
+  }
+  const converted = await converter.convertObj(fileData, {});
+  return { ...file, fileData: converted.openapi };
+};
+
+const sanitizeSchemaRefs = (pathSpec, renamedSchemas) => {
+  const text = JSON.stringify(pathSpec);
+  const sanitizedText = renamedSchemas.reduce((accu, { key, sanitizedKey }) => accu.replaceAll(`/schemas/${key}"`, `/schemas/${sanitizedKey}"`), text);
+  return JSON.parse(sanitizedText);
+};
+
+const repairPathSpec = (pathSpec, service, versionSuffix) => {
+  const text = JSON.stringify(pathSpec);
+  const sanitizedText = text.replaceAll(`"summary":`, `"description":`);
```

Comment:
I do not understand: why are we doing this replacement?

Reply:
some of the rules require a `description` but `summary` seems optional 🤷‍♂️